src/test/test-dlopen-so.c

```diff
@@ -27,76 +27,113 @@ static int run(int argc, char **argv) {
          * around yet. */
 
 #if HAVE_LIBIDN2 || HAVE_LIBIDN
-        assert_se(dlopen_idn() >= 0);
+        ASSERT_OK(dlopen_idn());
+#else
+        ASSERT_ERROR(dlopen_idn(), EOPNOTSUPP);
```

Comment:
Maybe, better to introduce `ASSERT_DLOPEN(expr, cond)` macro or helper function?